### PR TITLE
Revise signed payload repository

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
A few improvements made as part of https://github.com/digital-asset/daml/issues/8634.

- revised signed payload repository to hold more relevant data on the signed payload
- more specific types when handling bytes and strings

changelog_begin
changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: runtime-components/non-repudiation/src/main/scala/com/daml/nonrepudiation/SignedPayload.scala [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      algorithm: AlgorithmString,
+      fingerprint: FingerprintBytes,
+      payload: PayloadBytes,
+      signature: SignatureBytes,
+  ): SignedPayload =
+    new SignedPayload(algorithm, fingerprint, payload, signature)
+
+}
+
+// Purposefully not a case class because it contains mutable arrays
```
Comment (cocreature) [inline runtime-components/non-repudiation/src/main/scala/com/daml/nonrepudiation/SignedPayload.scala:18]: Do we actually need to carry mutable arrays through everything or could we wrap it in `ArraySeq.unsafeWrapArray` once at the library border and then stop worrying about it?

**Review by cocreature — APPROVED** [review]
Nice, thank you!

File: runtime-components/non-repudiation/src/main/scala/com/daml/nonrepudiation/SignedPayload.scala (hunk 1 above)
Comment (stefanobaghino-da) [reply runtime-components/non-repudiation/src/main/scala/com/daml/nonrepudiation/SignedPayload.scala:18]: That's a very good point, thanks!

Comment (stefanobaghino-da) [reply runtime-components/non-repudiation/src/main/scala/com/daml/nonrepudiation/SignedPayload.scala]: Addressed by https://github.com/digital-asset/daml/pull/8709/commits/a5e1b941f3408e9456d6d6a183b382ada2b2cae3

**Pull request merged**